job-service-contract/src/main/resources/com/hpe/caf/services/job/swagger.yaml

```diff
@@ -335,6 +337,41 @@ definitions:
         example:
           "tag:4": "4"
           owner: "bob"
+  expirationPolicy:
+    type: object
+    properties:
+      "Active":
+        $ref: "#/definitions/policy"
+      "Completed":
+        $ref: "#/definitions/policy"
+      "Failed":
+        $ref: "#/definitions/policy"
+      "Cancelled":
+        $ref: "#/definitions/policy"
+      "Waiting":
+        $ref: "#/definitions/policy"
+      "Paused":
+        $ref: "#/definitions/policy"
+      "Default":
+        $ref: "#/definitions/policy"
+    description: The expiration policy to be applied on the job
+  policy:
+    type: object
+    properties:
+      expiryTime:
+        type: string
+        format: date-time
```

Comment:
Declaring this as a `date-time` would mean that it could only accept exact times.  I could imagine that we might want to pass an exact expiry time but equally I think we should also support accepting an expiry expression like `lastUpdateTime+P1D`.

Reply:
used ´pattern: '^((lastUpdateDate)|(createDate))(\+P)([1-9]\d?|[12]\d{2}|3[0-5]\d|36[0-5])D$'´ 
action taken between 1 and 365 days. We can change that, I just thought we might need a max duration

By the way, I noticed that the pattern is not being passed to the classes. It has been raised some time ago [here](https://github.com/swagger-api/swagger-codegen/issues/7980) but I don't think that anyone works on it.

The way the classes are manually copied over, we can easily add it (it hasn't been done so far apparently)

I made a test with openAPI but same result -> restored back to Swagger 2 as the upgrade is not part of that Jira

Reply:
There is no need for a maximum duration.  There is a need to be able to specify less than a day.

Reply:
> By the way, I noticed that the pattern is not being passed to the classes. It has been raised some time ago here but I don't think that anyone works on it.

Yes the tooling support can vary quite a bit depends on what features you use.

> The way the classes are manually copied over, we can easily add it (it hasn't been done so far apparently)

Best just to enforce it in the implementation.

> I made a test with openAPI but same result -> restored back to Swagger 2 as the upgrade is not part of that Jira

Thank you.

Reply:
Added options for hour, minute and year in addition to day